File: WillMoveToOwnRepo/ProjectorRays/src/ProjectorRays.DotNet/director/Chunks/RaysInitialMapChunk.cs
using Microsoft.Extensions.Logging;
using ProjectorRays.Common;
using ProjectorRays.Director;

namespace ProjectorRays.director.Chunks;

public class RaysInitialMapChunk : RaysChunk
{
    public uint Version;
    public uint MmapOffset;

    public uint DirectorVersion { get; private set; }

    public RaysInitialMapChunk(RaysDirectorFile? dir) : base(dir, ChunkType.InitialMapChunk)
    {
        Writable = true;
    }

    public override void Read(ReadStream stream)
    {
        Version = stream.ReadUint32();
        MmapOffset = stream.ReadUint32();
        DirectorVersion = stream.ReadUint32(); 

        stream.Skip(8); // Skip remaining unused bytes
        Dir!.Logger.LogInformation($"InitialMapChunk: Version={Version}, MmapOffset={MmapOffset}, DirectorVersion={DirectorVersion}");

    }

    public override void WriteJSON(RaysJSONWriter json)
    {
        json.StartObject();
        json.WriteField("version", Version);
        json.WriteField("mmapOffset", MmapOffset);
        json.WriteField("directorVersion", DirectorVersion);
        json.EndObject();
    }
}

File: WillMoveToOwnRepo/ProjectorRays/src/ProjectorRays_Python/director/chunks/rays_initial_map_chunk.py
from .rays_chunk import RaysChunk, ChunkType
from ...common.stream import ReadStream
from ..rays_subchunk import MemoryMapEntry

class RaysInitialMapChunk(RaysChunk):
    def __init__(self, dir=None):
        super().__init__(dir, ChunkType.InitialMapChunk)
        self.version = 0
        self.mmap_offset = 0
        self.director_version = 0

    def read(self, stream: ReadStream):
        self.version = stream.read_uint32()
        self.mmap_offset = stream.read_uint32()
        self.director_version = stream.read_uint32()
        stream.skip(8)
        if self.dir:
            self.dir._logger.info(
                "InitialMapChunk: Version=%d, MmapOffset=%d, DirectorVersion=%d",
                self.version,
                self.mmap_offset,
                self.director_version,
            )

    def write_json(self, writer):
        writer.start_object()
        writer.write_key('version'); writer.write_val(self.version)
        writer.write_key('mmapOffset'); writer.write_val(self.mmap_offset)
        writer.write_key('directorVersion'); writer.write_val(self.director_version)
        writer.end_object()
